src/Stratis.Bitcoin.Features.MemoryPool/Interfaces/IMempoolRule.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Stratis.Bitcoin.Features.MemoryPool.Interfaces
+{
+    /// <summary>
+    /// Rule to be checked before a transaction is added to the mempool.
+    /// </summary>
+    public interface IMempoolRule
+    {
+        /// <summary>
+        /// Check that the transaction meets certain criteria before being added to mempool.
+        /// If it doesn't, a <see cref="ConsensusErrorException" /> will be thrown.
+        /// </summary>
+        /// <param name="ruleContext">Current rule context.</param>
```

Comment:
These two params can be slightly better described to highlight their differences.